### DISCUSSION TITLE
how to pass -Djava.library.path=... to quarkus dev

### DISCUSSION THREAD

**Discussion by kurtseebauer** [quarkusio/quarkus#36750] [Q&A]
Hi all,

I need to start my application with a native library, using the `-Djava.library.path` parameter. the `java -jar` direct start works fine, but I also want to start the dev server with gradle. `quarkus dev` or `quarkus dev -Djava.library.path=...` leads to 

```
Caused by: java.lang.UnsatisfiedLinkError: no getdrgjni in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
```
Is there any way I could make the dev server aware of my library?

Thank you!
--Kurt

**Comment by mschorsch** [comment]
I think you have to configure this option in the quarkusDev task in your build.gradle(.kts).

https://quarkus.io/guides/gradle-tooling#dev-mode

**Comment by mschorsch** [comment]
The order of the properties on the command line is also relevant for the gradle cli (https://docs.gradle.org/current/userguide/command_line_interface.html). I am not sure if this is also true for system properties.

You could try `gradlew -Djava.library.path=... quarkusDev` instead of `gradlew quarkusDev -Djava.library.path=...`.

I think it won't work either but you can give it a try. 😄

**Comment by kurtseebauer** [comment]
Hi @mschorsch , thanks for the pointer. For some reason passing the argument to `gradlew quarkusDev` does not work for java.library.path as it does for other arguments, also there is no equivalent env variable like `JAVA_LIBRARY_PATH`. 

But, setting `LD_LIBRARY_PATH` to include my binary library seems to do the trick. 
For future reference:
```
LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/my/binary/libraries quarkus dev
```
